app/src/main/java/com/orange/ouds/app/ui/utilities/Code.kt

```diff
@@ -0,0 +1,255 @@
+
+    override fun format(context: Context): String {
+        val isMultiline = isMultiline && elements.isNotEmpty()
+        val elementSeparator = if (isMultiline) "\n" else " "
+        val lambda = elements.lastOrNull()?.asOrNull<Argument<*>>()?.value?.asOrNull<Lambda>().takeIf { trailingLambda }
+        val elements = if (lambda != null) elements.dropLast(1) else elements
+        val formattedElements = elements.joinToString(elementSeparator) { codeFormattable ->
+            val elementSuffix = if (codeFormattable is Argument<*>) "," else ""
+            "${codeFormattable.format(context)}$elementSuffix"
+        }
+            .removeSuffix(",")
+            .run { if (isMultiline) prependIndent(CODE_INDENT) else this }
+        val parenthesisSeparator = if (isMultiline) "\n" else ""
+
+        val formattedLambda = if (lambda != null) " ${lambda.format(context)}" else ""
+
+        return "$name($parenthesisSeparator$formattedElements$parenthesisSeparator)$formattedLambda"
+    }
+
+    @CodeDslMarker
+    class Builder {
+
+        var name: String by Delegates.notNull()
+
+        @PublishedApi
+        internal var elements: MutableList<CodeFormattable> = mutableListOf()
+
+        var isMultiline = true
+
+        var trailingLambda = false
+
+        inline fun <reified T> typedArgument(name: String?, value: T) {
+            elements.add(Argument(name, value, T::class.java))
+        }
+
+        fun rawArgument(name: String?, value: String) = typedArgument(name, RawArgumentValue(value))
+
+        fun lambdaArgument(name: String?, init: Code.Builder.() -> Unit = {}) {
+            val code = Code.Builder().apply(init).build()
+            typedArgument(name, Lambda(code))
+        }
+
+        fun functionCallArgument(name: String?, functionName: String, init: Builder.() -> Unit = {}) {
+            val builder = Builder().apply {
+                this.name = functionName
+                init()
+            }
+            val functionCall = builder.build()
+            typedArgument(name, functionCall)
+        }
+
+        inline fun <reified T> constructorCallArgument(name: String?, noinline init: Builder.() -> Unit = {}) {
+            functionCallArgument(name, T::class.java.nestedName, init)
+        }
+
+        fun comment(text: String, init: Comment.Builder.() -> Unit = {}) {
+            val comment = Comment.Builder()
+                .apply {
+                    this.text = text
+                    init()
+                }
+                .build()
+            elements.add(comment)
+        }
+
+        fun build() = FunctionCall(name, elements, isMultiline, trailingLambda)
+    }
+}
+
+data class Argument<T>(val name: String?, val value: T, val clazz: Class<T>) : CodeFormattable {
+
+    override fun format(context: Context): String {
+        val valueString = when (value) {
+            is Float -> "${value}F"
```

Comment:
`Float` is more represented with a lower case `f` no?

Reply:
That's what I thought too. But if you write `val a: Float = 1.0`, then Android Studio suggests to change this code to `val a: Float = 1.0F`.